### PR TITLE
refactor: simplify the call in lib.picocolors

### PR DESCRIPTION
**Pull request by huozhi**
Simplify the conditional formatter call in `lib/picocolors`

Closes NEXT-2018

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details open>
<summary><strong>Default Build</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | vercel/next.js refactor/formatter | Change |
| - | - | - | - |
| buildDuration | 12.8s | 12.8s | N/A |
| buildDurationCached | 7.1s | 6.2s | N/A |
| nodeModulesSize | 200 MB | 200 MB | N/A |
| nextStartRea..uration (ms) | 428ms | 430ms | N/A |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js refactor/formatter | Change |
| - | - | - | - |
| 193.HASH.js gzip | 181 B | 182 B | N/A |
| 3f784ff6-HASH.js gzip | 53.3 kB | 53.3 kB | N/A |
| 433-HASH.js gzip | 28.6 kB | 28.6 kB | N/A |
| framework-HASH.js gzip | 45.2 kB | 45.2 kB | ✓ |
| main-app-HASH.js gzip | 240 B | 242 B | N/A |
| main-HASH.js gzip | 31.7 kB | 31.8 kB | N/A |
| webpack-HASH.js gzip | 1.7 kB | 1.7 kB | N/A |
| Overall change | 45.2 kB | 45.2 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js refactor/formatter | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js refactor/formatter | Change |
| - | - | - | - |
| _app-HASH.js gzip | 194 B | 195 B | N/A |
| _error-HASH.js gzip | 183 B | 181 B | N/A |
| amp-HASH.js gzip | 504 B | 502 B | N/A |
| css-HASH.js gzip | 321 B | 321 B | ✓ |
| dynamic-HASH.js gzip | 2.5 kB | 2.5 kB | N/A |
| edge-ssr-HASH.js gzip | 255 B | 253 B | N/A |
| head-HASH.js gzip | 350 B | 349 B | N/A |
| hooks-HASH.js gzip | 369 B | 369 B | ✓ |
| image-HASH.js gzip | 4.28 kB | 4.28 kB | N/A |
| index-HASH.js gzip | 255 B | 256 B | N/A |
| link-HASH.js gzip | 2.61 kB | 2.61 kB | ✓ |
| routerDirect..HASH.js gzip | 312 B | 311 B | N/A |
| script-HASH.js gzip | 385 B | 383 B | N/A |
| withRouter-HASH.js gzip | 307 B | 308 B | N/A |
| 1afbb74e6ecf..834.css gzip | 106 B | 106 B | ✓ |
| Overall change | 3.4 kB | 3.4 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js refactor/formatter | Change |
| - | - | - | - |
| _buildManifest.js gzip | 483 B | 484 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js refactor/formatter | Change |
| - | - | - | - |
| index.html gzip | 528 B | 527 B | N/A |
| link.html gzip | 541 B | 539 B | N/A |
| withRouter.html gzip | 525 B | 522 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Edge SSR bundle Size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js refactor/formatter | Change |
| - | - | - | - |
| edge-ssr.js gzip | 93.8 kB | 93.8 kB | N/A |
| page.js gzip | 148 kB | 148 kB | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Middleware size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js refactor/formatter | Change |
| - | - | - | - |
| middleware-b..fest.js gzip | 622 B | 622 B | ✓ |
| middleware-r..fest.js gzip | 151 B | 151 B | ✓ |
| middleware.js gzip | 37.4 kB | 37.4 kB | N/A |
| edge-runtime..pack.js gzip | 1.92 kB | 1.92 kB | ✓ |
| Overall change | 2.7 kB | 2.7 kB | ✓ |

</details>

<details>
<summary><strong>Next Runtimes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js refactor/formatter | Change |
| - | - | - | - |
| app-page-exp...dev.js gzip | 169 kB | 169 kB | N/A |
| app-page-exp..prod.js gzip | 95.2 kB | 95.2 kB | ✓ |
| app-page-tur..prod.js gzip | 95.8 kB | 95.8 kB | N/A |
| app-page-tur..prod.js gzip | 90.4 kB | 90.4 kB | N/A |
| app-page.run...dev.js gzip | 142 kB | 142 kB | N/A |
| app-page.run..prod.js gzip | 89.8 kB | 89.8 kB | N/A |
| app-route-ex...dev.js gzip | 24.1 kB | 24.1 kB | N/A |
| app-route-ex..prod.js gzip | 16.7 kB | 16.7 kB | N/A |
| app-route-tu..prod.js gzip | 16.7 kB | 16.7 kB | N/A |
| app-route-tu..prod.js gzip | 16.3 kB | 16.3 kB | N/A |
| app-route.ru...dev.js gzip | 23.5 kB | 23.5 kB | N/A |
| app-route.ru..prod.js gzip | 16.3 kB | 16.3 kB | N/A |
| pages-api-tu..prod.js gzip | 9.38 kB | 9.38 kB | ✓ |
| pages-api.ru...dev.js gzip | 9.65 kB | 9.65 kB | ✓ |
| pages-api.ru..prod.js gzip | 9.37 kB | 9.37 kB | ✓ |
| pages-turbo...prod.js gzip | 21.9 kB | 21.9 kB | N/A |
| pages.runtim...dev.js gzip | 22.6 kB | 22.6 kB | N/A |
| pages.runtim..prod.js gzip | 21.9 kB | 21.9 kB | N/A |
| server.runti..prod.js gzip | 49.5 kB | 49.5 kB | N/A |
| Overall change | 124 kB | 124 kB | ✓ |

</details>

<details>
<summary><strong>Diff details</strong></summary>

<details>
<summary>Diff for <strong>page.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>edge-ssr.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page-exp..ntime.dev.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page-exp..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page-tur..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page-tur..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page.runtime.dev.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page.runtime.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-route-ex..ntime.dev.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-route-ex..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-route-tu..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-route-tu..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-route.runtime.dev.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-route.ru..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>pages-turbo...time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>pages.runtime.dev.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>pages.runtime.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>server.runtime.prod.js</strong></summary>

Diff too large to display
</details>

</details>

</details>
Commit: 6c85b249adf77beb998dd7e7f7930306528d91c0

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 6c85b249adf77beb998dd7e7f7930306528d91c0

`pnpm test-dev test/e2e/app-dir/navigation/navigation.test.ts` 
- app dir - navigation > query string > useParams identity between renders > should be stable in pages

<details>
<summary>Expand output</summary>

  ● app dir - navigation › query string › useParams identity between renders › should be stable in pages

    TIMED OUT: success

    fail

    undefined

      626 |
      627 |   if (hardError) {
    > 628 |     throw new Error('TIMED OUT: ' + regex + '\n\n' + content + '\n\n' + lastErr)
          |           ^
      629 |   }
      630 |   return false
      631 | }

      at check (lib/next-test-utils.ts:628:11)
      at runTests (e2e/app-dir/navigation/navigation.test.ts:88:11)
      at Object.<anonymous> (e2e/app-dir/navigation/navigation.test.ts:113:11)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

File: packages/next/src/lib/picocolors.ts [hunk 1]
```diff
@@ -47,35 +47,32 @@ const formatter =
       : open + string + close
   }
 
+const color = (...args: Parameters<typeof formatter>) =>
+  enabled ? formatter(...args) : String
```
Comment (styfle) [inline packages/next/src/lib/picocolors.ts:51]: Could we avoid a new function and instead move the check for `enabled` boolean into `formatter`?

Such as:

```
const formatter = {
  if (!enabled) return String
  ...
}
```

**Review by styfle — APPROVED** [review]
Thanks!

**Pull request merged**